src/main/java/Deadline.java

```diff
@@ -0,0 +1,39 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    LocalDate taskBy;
+
+    Deadline(String taskName, LocalDate taskBy) {
+        super(taskName);
+        this.taskBy = taskBy;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]"
+                + super.toString()
+                + " (by: "
+                + this.taskBy.format(DateTimeFormatter.ofPattern("MMM dd yyyy"))
+                + ")";
+    }
+
+    @Override
+    public String toFileFormat() {
+        return "D|" + (this.isDone ? "1" : "0") + "|" + this.taskName + "|" + this.taskBy;
+    }
+
+    /**
+     * Loads the file format String representation of Deadline task.
+     *
+     * @return Deadline loaded from file format representation
+     */
+    static Deadline fromFileFormat(String fileFormatString) {
```

Comment:
Same as hanweic53 has commented, it could be possible to modify this naming a bit so that it might sound clearer :) The descriptions for methods are quite clear though! Nice attempt!